include/event_logger.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "constants.hpp"
+#include "types.hpp"
+
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+namespace vpd
+{
+/**
+ * @brief Class to support logging events.
```

Comment:
"support" What is meant by support here?

Reply:
updated.